osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -63,6 +66,10 @@ public class BeatmapManager
 
         private readonly BeatmapStore beatmaps;
 
+        private readonly APIAccess api;
+
+        private readonly Dictionary<int?, DownloadBeatmapSetRequest> downloadsMap;

```

Comment:
two things:
- having a nullable *anything* as a key is very very not good.
- it's unclear what the `int` is here. consider either just making this a list of `DownloadBeatmapSetRequest` (and exposing the `BeatmapSetInfo` from inside it to do lookups) or making the key a `BeatmapSetInfo`

Reply:
based on the use case here, i'd probably just use a list. we don't really need the extra lookup efficiency because this list should never get that large.